pkg/functions/openfaas/driver.go

```diff
@@ -227,21 +251,22 @@ func (d *ofDriver) GetRunnable(e *functions.FunctionExecution) functions.Runnabl
 		case 200:
 			resBytes, err := ioutil.ReadAll(res.Body)
 			if err != nil {
-				return nil, errors.Errorf("cannot read result from OpenFaaS on URL: %s %s", d.gateway, err)
+				return nil, &systemError{errors.Errorf("cannot read result from OpenFaaS on URL: %s %s", d.gateway, err)}
 			}
 			var out functions.Message
 			if err := json.Unmarshal(resBytes, &out); err != nil {
-				return nil, errors.Errorf("cannot JSON-parse result from OpenFaaS: %s %s", err, string(resBytes))
+				return nil, &outputError{errors.Errorf("cannot JSON-parse result from OpenFaaS: %s %s", err, string(resBytes))}
```

Comment:
This one is a system error too: it's the function runtime responsible for JSON-encoding the response, not the function.